x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/authc/AuthenticationTestHelper.java

```diff
@@ -233,6 +237,43 @@ public static String randomInternalRoleName() {
         );
     }
 
+    public static RemoteAccessAuthentication randomRemoteAccessAuthentication() {
```

Comment:
The class name `RemoteAccessAuthentication` is a bit confusing, in retrospect. It's ambiguous if it refers to the header sent by the querying cluster, or a fulfilling-side `remote_access` authentication instance. I work on some renaming in a follow up PR (`RemoteAccessAuthenticationHeaders` to designate the headers sent by the QC or something along these lines).

Reply:
Yeah I think rename might help. How about making it more specific to something like `QueryClusterAuthenticationInformation`?

Reply:
I'm thinking `RemoteClusterAuthenticationInformation` to avoid Query or QueryingCluster -- to me these terms are largely for internal use and even though they've made it into PR descriptions and some test code I slightly prefer sticking to RemoteCluster in prod code, unless it's really ambiguous.